### ISSUE TITLE
[DataGrid] The density selector isn't accessible

### ISSUE THREAD

**Issue by oliviertassinari** [mui-org/material-ui-x#881]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

<img width="193" alt="Capture d’écran 2021-01-20 à 00 37 05" src="https://user-images.githubusercontent.com/3165635/105107128-a7766000-5ab7-11eb-8fac-89cb5eca4bd6.png">

## Current Behavior 😯

1. The button has `aria-haspopup="true"`
1. The button doesn't have `aria-labelledby`
1. The button doesn't have `aria-expanded="true"`
1. Opening the popup doesn't move the focus to it
1. <kbd>Arrow Down</kbd> doesn't allow to navigate the options
1. <kbd>Escape</kbd> doesn't closes the popup
1. The ul has a role "menu"
1. The ul has no aria-labelledby

## Expected Behavior 🤔

1. The button has `aria-haspopup="listbox"`
1. The button has a `aria-labelledby` that points to the listbox
1. The button has `aria-expanded="true"`
1. Opening the popup moves the focus to it
1. <kbd>Arrow Down</kbd> allows to navigate the options
1. <kbd>Escape</kbd> closes the popup
1. The ul has a role "listbox"
1. The ul has aria-labelledby

The pattern: https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html

## Steps to Reproduce 🕹

Steps:

1. https://master--material-ui-x.netlify.app/components/data-grid/columns/#column-selector

## Context 🔦

⚠️ The behavior required for the density selector is different from the one needed for the column menu (#882).

## Your Environment 🌎

`@material-ui/data-grid@v4.0.0-alpha.17`

**Comment by m4theushw** [comment]
@chloesun 

```jsx
<DataGrid density="compact" />
```

**Comment by chloesun** [comment]
@m4theushw thank you so much!
